numba_dppy/ocl/ocldecl.py

```diff
@@ -164,7 +165,7 @@ def typer(shape, dtype):
             ndim = parse_shape(shape)
             nb_dtype = parse_dtype(dtype)
             if nb_dtype is not None and ndim is not None:
-                return types.Array(
+                return DPPYArray(
                     dtype=nb_dtype,
                     ndim=ndim,
                     layout="C",
```

Comment:
Do we envision supporting strides and non-C contiguous data layouts?

Reply:
Moved to https://github.com/IntelPython/numba-dppy/issues/363.